src/biclustering_impl.h

```diff
@@ -376,17 +275,14 @@ class ConvexBiClustering {
   Eigen::Index nzeros_row; // Fusion counts
   Eigen::Index nzeros_col;
 
-  // The DLPA (on which CBASS is based) adds two auxiliary variables -- P & Q --
-  // with the same dimensions as the optimization variable, initialized to zero
-  Eigen::MatrixXd P;
-  Eigen::MatrixXd Q;
+  // Precomputed products that are reused in U-update
+  Eigen::MatrixXd DDT_col; // D_col * D_col^T
+  Eigen::MatrixXd DTD_row; // D_row^T * D_row
```

Comment:
Can we set these as `const` ?  Const members are a bit nasty to set up - you have to put them in the initializer (see how things like `X` get initialized), but I think it's better semantics.

Reply:
Also slightly above this - can't comment since it's not a change. The `row_primal_solver` isn't used any more, so can we delete it?